cmd/chisel/cmd_cut.go

```diff
@@ -82,29 +82,72 @@ func (cmd *cmdCut) Execute(args []string) error {
 		return err
 	}
 
+	archives, err := openArchives(release, cmd.Arch)
+	if err != nil {
+		return err
+	}
+
+	report, err := slicer.Run(&slicer.RunOptions{
+		Selection:   selection,
+		PkgArchives: archives,
+		TargetDir:   cmd.RootDir,
+	})
+	if err != nil {
+		return err
+	}
+
+	manifestSlices := locateManifests(selection.Slices)
+	if len(manifestSlices) > 0 {
+		pkgInfo, err := gatherPackageInfo(selection, archives)
```

Comment:
Hmm.. we can do that of course. But the number of elements in the final array would be more than what we need. Because the `pkgArchives` is a map for all packages to their respective archvies. Although we do need only the selected packages there.

Reply:
[IMPORTANT] Then my bad for not spotting it earlier, we have to change this. In the previous version of the code we were only calling `.Exists` for packages in the selection. Let's do the same here please.

If we split the functions in https://github.com/letFunny/chisel/pull/11#discussion_r1574285774 then you can pass the selection and []*setup.Archive to the second one.

Reply:
Done in 26385d21eac1ca9150049fe58519374f2c1861d9.